models/userModel.js

```diff
@@ -0,0 +1,52 @@
+const mongoose = require("mongoose");
+const validator = require("validator");
+const bcrypt = require("bcrypt");
+const userSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: [true, "Pls, tell us your name"],
+  },
+  email: {
+    type: String,
+    required: [true, "Please provide your email"],
+    unique: true,
+    lowercase: true,
+    validate: [validator.isEmail, "Please provide a valid email"],
+  },
+  photo: {
+    type: String,
+  },
+  password: {
+    type: String,
+    required: [true, "Please provide a password"],
+    minLength: 8,
+    select: false,
+  },
+  passwordConfirm: {
+    type: String,
+    required: [true, "Confirm Password"],
+    minLength: 8,
+    validate: {
```

Comment:
Also, I could see you are validating something here but that doesn't mean you shouldn't have the validation done on the request object itself before you even get to Crete the db model

Reply:
okay, will do that